Manila: add Get for share-access-rules API

gman0 opened the pull request:
This PR adds `Get` for `share-access-rules` Manila API.

Part of https://github.com/gophercloud/gophercloud/issues/2495

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/manila/blob/stable/yoga/manila/db/sqlalchemy/api.py#L2591-L2601
https://github.com/openstack/manila/blob/stable/yoga/manila/api/v2/share_accesses.py#L39-L45
https://github.com/openstack/manila/blob/stable/yoga/manila/api/views/share_accesses.py#L50-L62

gman0 commented:
Ready for reviews. PTAL @mandre @gouthampacha, thanks!

mandre commented:
I'll try my best to clear my backlog of reviews before the end of the week, sorry for the time it takes.

gman0 commented:
Thanks, @mandre!

gman0 commented:
I've just added an acceptance test but it's failing:

```
=== RUN   TestShareAccessRulesGet
    shareaccessrules_test.go:29: Unable to grant access to share 70959eca-bcb4-4999-afaf-3e371e044ac1: Resource not found: [POST http://10.1.0.220/share/v2/shares/70959eca-bcb4-4999-afaf-3e371e044ac1/action], error message: {"itemNotFound": {"code": 404, "message": "The resource could not be found."}}
    shares.go:98: Deleted share: 70959eca-bcb4-4999-afaf-3e371e044ac1
--- FAIL: TestShareAccessRulesGet (4.38s)

```

https://github.com/gophercloud/gophercloud/actions/runs/3429349022/jobs/5714846083#step:5:1097

mandre commented on openstack/sharedfilesystems/v2/shareaccessrules/results.go:
```diff
@@ -0,0 +1,65 @@
+	type tmp ShareAccess
+	var s struct {
+		tmp
+		CreatedAt gophercloud.JSONRFC3339MilliNoZ `json:"created_at"`
```
According to [the doc](https://docs.openstack.org/api-ref/shared-file-system/?expanded=list-access-rules-deprecated-detail#id610), the date is returned in ISO 8601 format. Was JSONRFC3339MilliNoZ perhaps an old date format used by Manila?

gman0 replied:
It seems RFC3339 offers ISO 8601 interoperability: https://www.rfc-editor.org/rfc/rfc3339. Also other structs seem to use this JSONRFC3339MilliNoZ where the docs mention ISO 8601... I'm not an expert in this area

gouthampacha replied:
I don't get the question perhaps, all date and timestamp fields from openstack services use this datatype within the gophercloud SDK:

https://github.com/gophercloud/gophercloud/blob/513734676e6495f6fec60e7aaf1f86f1ce807428/results.go#L308

mandre replied:
I've reported https://github.com/gophercloud/gophercloud/issues/2507 for the date parsing issue if anyone wants to take a crack at it.

mandre commented:
> I've just added an acceptance test but it's failing:
> 
> ```
> === RUN   TestShareAccessRulesGet
>     shareaccessrules_test.go:29: Unable to grant access to share 70959eca-bcb4-4999-afaf-3e371e044ac1: Resource not found: [POST http://10.1.0.220/share/v2/shares/70959eca-bcb4-4999-afaf-3e371e044ac1/action], error message: {"itemNotFound": {"code": 404, "message": "The resource could not be found."}}
>     shares.go:98: Deleted share: 70959eca-bcb4-4999-afaf-3e371e044ac1
> --- FAIL: TestShareAccessRulesGet (4.38s)
> ```
> 
> https://github.com/gophercloud/gophercloud/actions/runs/3429349022/jobs/5714846083#step:5:1097
> 
> I'll have a look ASAP, but if anyone has ideas what's wrong please let me know.

I haven't looked into the details but according to https://github.com/gophercloud/gophercloud/pull/1795#issuecomment-565005555, setting the microversion to 2.49 helps.

gouthampacha commented on acceptance/openstack/sharedfilesystems/v2/shareaccessrules_test.go:
```diff
@@ -0,0 +1,48 @@
+		t.Fatalf("Unable to create a shared file system client: %v", err)
+	}
+
+	client.Microversion = "2.49"
```
this API should be available from "2.45": https://docs.openstack.org/api-ref/shared-file-system/?expanded=list-access-rules-deprecated-detail,list-shares-detail#share-access-rules-since-api-v2-45

gouthampacha commented:
Thanks @gman0 - the code LGTM!

mandre reviewed (commented):
Not necessarily for this PR since the issue seems to be more generalized in the sharedfilesystems module, and possibly elsewhere, but the date parsing seems fragile to me and it would be nice to clean it at some point.
